scripts/generate/templates/Shim.mustache

```diff
@@ -1,7 +1,9 @@
 {-|
 Module: Database.Persist.Monad.Shim
 
-Defines all the @persistent@ functions lifted into 'MonadSqlQuery'.
+Defines all the @persistent@ functions lifted into 'MonadSqlQuery', and
```

Comment:
Let me think about this for a sec. Personally, I'd like to keep Shim.hs very simple (i.e. Shim refers to the shimmed functions, not shimming the entire Database.Persist.Sql module).

Also, just some clarification on the motivation: this is only an issue when universally importing both Database.Persist.Sql and Database.Persist.Monad.Shim unqualified, right? IMO you shouldnt be doing that, but you do you.

I would be much more amenable if this added a separate module like `Database.Persist.Sql.Shim` that re-exported `Database.Persist.Sql` and `Database.Persist.Monad.Shim`. What do you think?

Reply:
>Also, just some clarification on the motivation: this is only an issue when universally importing both Database.Persist.Sql and Database.Persist.Monad.Shim unqualified, right?

Not quite. It is an issue wherever the functions from `Database.Persist.Monad` and `Database.Persist.Sql` (or simply `Database.Persist`), are in the same namespace - for example, if they're imported with the same qualifier.

You might also feel that sharing a qualifier across these modules is a mistake, but for me it's a massive convenience - I didn't want to force devs to manually juggle different qualifiers for the "correct" functions (because we want to be using the mtl-style ones). Having to explicitly import from `Database.Persist` or `Database.Persist.Sql` whenever I wanted access to types or helpers was also annoying.

For context, I dropped `persistent-mtl` (or more accurately, a very opinionated fork of it, found [here](https://github.com/SupercedeTech/persistent-mtl/tree/refactor)) into a codebase with lots of existing usage of `persistent`. I find the package massively helpful, but I needed to be pragmatic in making the migration simple - so I made `Database.Persist.Monad` a drop-in replacement.

>I would be much more amenable if this added a separate module like Database.Persist.Sql.Shim that re-exported Database.Persist.Sql and Database.Persist.Monad.Shim. What do you think?

I think this would make sense as well.

Reply:
Does this need an upstream change? Couldnt you just add a module in your project that re-exports the things? It would be a bit of copy-paste, but it would only be a one-time thing. It seems like that would be sufficient if its mostly a thing to ease migration. Just speaking hypothetically

Reply:
>Does this need an upstream change?

By no means, no. We're able to use our fork internally without issue.

But since we're benefiting so much from the package, I wanted to see what we could contribute back. Most of the other changes we've made are motivated by internal requirements, and probably don't make sense to upstream.

Reply:
Sorry, I mean like, couldnt you manually reexport things in your project without needing to edit persistent-mtl (a fork or otherwise)?

Reply:
In principle, yes.